### PR TITLE
Python binding improvement

### PR DESCRIPTION
**Pull request by maxvyaznikov**
Completely reworked python binding. There were many unneeded steps into API.

### CODE REVIEW

**Comment by linas** [comment]
Wow.

File: bindings/python/linkgrammar.py [hunk 1]
```diff
@@ -81,210 +38,306 @@ def __init__(self, verbosity = 0,
...
+    def __del__(self):
+        if hasattr(self, '_obj'):
+            clg.sentence_delete(self._obj)
+            del self._obj
+
+    def split(self):
+        return clg.sentence_split(self._obj)
+
+    def __len__(self):
+        return clg.sentence_length(self._obj)
```
Comment (linas) [inline bindings/python/linkgrammar.py]: This is technically incorrect, and will need to change. The actual length of the sentence may vary, depending on the linkage.   So, for example: "We saw the seasand beach"  which is clearly mis-spelled .. but is it "we saw the seas and beach" (6 words) or is it "we saw the seasoned beach" (5 words)?   These are relatively rare, but the new tokenizer design allows these.

Similarly, the sentence no longer has a unique splitting into words, so the split() above cannot be used.

File: bindings/python/linkgrammar.py [hunk 2]
```diff
@@ -81,210 +38,306 @@ def __init__(self, verbosity = 0,
...
+
+    def parse(self):
+        n = clg.sentence_parse(self._obj, self.parse_options._obj)
+        for i in xrange(n):
+            yield Linkage(i, self, self.parse_options._obj)
```
Comment (linas) [inline bindings/python/linkgrammar.py]: Hmm, also, should probably not create linkages until they are actually wanted. That will save some cpu time (not much, but some)

**Comment by linas** [comment]
Whoops .. I just merged this, and I realized I shouldn't have.   I was planning on having a version 5.2.5 first, which contains only bug fixes, and then have your code, and the new tokenizer, in version 5.3.0, sometime later.   Although I suspect that no one is actually using the python bindings, it still seems unfair to just change them without a somewhat bigger version bump.

**Comment by maxvyaznikov** [comment]
Yeah, I thought you wait unitil next release =)
And, yes, python binding was very dirty to serious usage.
I'll fix bugs above and make more tests.

**Comment by ampli** [comment]
Hello Max,
Please see my problem at #67.
I would appreciate your opinion on that.
Is there a way to run the tests without installing?

**Pull request merged**